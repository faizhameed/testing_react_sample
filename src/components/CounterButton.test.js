import { shallow } from "enzyme";
import React from "react";
import CounterButton from "./CounterButton";
import toJson from "enzyme-to-json";

it("expect to run CounterButton componen", () => {
  const mockColor = "red";
  const wrapper = shallow(<CounterButton color={mockColor} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

it("is correct increment function", () => {
  const mockColor = "red";
  const wrapper = shallow(<CounterButton color={mockColor} />);

  wrapper.find('[id="counter"]').simulate("click");
  expect(wrapper.state()).toEqual({ count: 1 });
  wrapper.find('[id="counter"]').simulate("click");
  expect(wrapper.state()).toEqual({ count: 2 });
  expect(wrapper.props().color).toEqual("red");
});
