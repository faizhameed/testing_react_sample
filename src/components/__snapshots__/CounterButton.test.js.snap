// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to run CounterButton componen 1`] = `
<button
  color="red"
  id="counter"
  onClick={[Function]}
>
  Count: 
  0
</button>
`;
